### PR TITLE
[Mailer] add ability to disable the TLS peer verification via DSN

### PR DESCRIPTION
**Pull request by Livda**
Documentation for the PR https://github.com/symfony/symfony/pull/35262

### CODE REVIEW

**Review by Nyholm — CHANGES_REQUESTED** [review]
> like if the SMTP server has a self signed certificate

You **should** still verify it. But if you are lazy like I am, then you are most likely to ignore it with this new settings =). 

I think this should be rephrased. 

-------

I agree that this should be a separate paragraph, because many SMTP related transports extends from `EstmpTransport` which brings this feature.

File: components/mailer.rst [hunk 1]
```diff
@@ -142,6 +142,17 @@ The mailer will start using the first transport and if it fails, it will retry
 the same delivery with the next transports until one of them succeeds (or until
 all of them fail).
 
+Disable TLS peer verification
```
Comment (Nyholm) [inline components/mailer.rst]: TLS peer verification

File: components/mailer.rst [hunk 2]
```diff
@@ -142,6 +142,17 @@ The mailer will start using the first transport and if it fails, it will retry
...
+-----------------------------
+
+Symfony's mailer supports TLS peer verification by default, but in some cases,
```
Comment (Nyholm) [inline components/mailer.rst]: SMTP tranports supports TLS peer verification by default, but in some cases, you may want to turn this off. Ie in a development environment.

File: components/mailer.rst [hunk 3]
```diff
@@ -142,6 +142,17 @@ The mailer will start using the first transport and if it fails, it will retry
...
 all of them fail).
 
+Disable TLS peer verification
+-----------------------------
+
+Symfony's mailer supports TLS peer verification by default, but in some cases, 
+this verification shouldn't be done, like if the SMTP server has a self signed 
+certificate.
+
+In such case, a parameter can be used in the DSN::
```
Comment (Nyholm) [inline components/mailer.rst]: In such case, add the `verify_peer` parameter to the DSN::

**Review by Nyholm — CHANGES_REQUESTED** [review]
Thank you. 

(Sorry for the multiple PR reviews)

File: components/mailer.rst (hunk 1 above)
Comment (OskarStark) [reply components/mailer.rst]: As this is a title, we prefer:
```suggestion
TLS Peer Verification
```

Please adjust the underline length accordingly 👍

**Comment by javiereguiluz** [comment]
@Livda thanks for this contribution (and for contributing the feature too). We did some rewords while merging and added the missing `versionadded:: 5.1` directive, but your original contribution was great.

Congrats on your first Symfony Docs contribution!

**Pull request merged**